src/server/game/Entities/Player/Player.cpp

```diff
@@ -16440,12 +16441,18 @@ bool Player::HasQuestForItem(uint32 itemid) const
 
         QuestStatusData const& q_status = qs_itr->second;
 
-        if (q_status.Status == QUEST_STATUS_INCOMPLETE)
+        if (q_status.Status == QUEST_STATUS_INCOMPLETE || q_status.Status == QUEST_STATUS_COMPLETE)
```

Comment:
Now that was a constructive comment. That is very easily fixed by adding a bool because I need to check for both, but yes, I see how that breaks other uses.